packages/zeego/src/menu/create-android-menu/index.android.tsx

```diff
@@ -1,356 +1,442 @@
+import { menuify } from '../display-names'
+import { filterNull } from '../filter-null'
 import type {
-  MenuCheckboxItemProps,
   MenuContentProps,
   MenuGroupProps,
-  MenuItemIconProps,
-  MenuItemImageProps,
-  MenuItemIndicatorProps,
   MenuItemProps,
   MenuItemSubtitleProps,
   MenuItemTitleProps,
-  MenuLabelProps,
   MenuRootProps,
-  MenuSeparatorProps,
   MenuTriggerItemProps,
   MenuTriggerProps,
+  MenuItemIconProps,
+  MenuCheckboxItemProps,
+  MenuSeparatorProps,
+  MenuItemImageProps,
+  MenuItemIndicatorProps,
+  MenuLabelProps,
 } from '../types'
-
-import { menuify } from '../display-names'
-
-import { Popover } from 'react-native-popper'
-import { pickChildren } from '../children'
-
-type MenuVisibleContext = {
-  isOpen: boolean
-  onClose: () => void
-  onShow: () => void
-  onOpenChange: (next: boolean) => void
-  closeRootMenu: () => void
-  onToggleOpen: () => void
-}
-
-const VirtualizedMenuContext = createContext(false)
-const useIsNestedMenu = () => useContext(VirtualizedMenuContext)
-
-const MenuVisibleContext = createContext<MenuVisibleContext>(null as any)
-
-const useMenuVisibleContext = () => useContext(MenuVisibleContext)
-
-const TriggerItem = menuify(
-  ({
-    style,
-    onFocus,
-    onBlur,
-    textValue,
-    disabled,
-    children,
-  }: MenuTriggerItemProps) => {
-    const { onToggleOpen } = useMenuVisibleContext()
-    return (
-      <Pressable
-        onPress={onToggleOpen}
-        style={style}
-        onFocus={onFocus}
-        onBlur={onBlur}
-        accessibilityLabel={textValue}
-        disabled={disabled}
-      >
-        {children}
-      </Pressable>
-    )
-  },
-  'TriggerItem'
-)
-
-function MenuProvider({ children }: { children: ReactNode }) {
-  const [isOpen, setOpen] = useState(false)
-
-  const parentContext = useMenuVisibleContext()
-
-  const isNestedMenu = !!parentContext
-
-  const closeRootMenu = parentContext?.closeRootMenu
-
-  return (
-    <MenuVisibleContext.Provider
-      value={useMemo(
-        () => ({
-          isOpen,
-          onShow: () => setOpen(true),
-          onClose: () => setOpen(false),
-          onOpenChange: setOpen,
-          closeRootMenu: closeRootMenu ?? (() => setOpen(false)),
-          onToggleOpen: () => setOpen((prev) => !prev),
-        }),
-        [isOpen]
-      )}
-    >
-      <VirtualizedMenuContext.Provider value={isNestedMenu}>
-        {children}
-      </VirtualizedMenuContext.Provider>
-    </MenuVisibleContext.Provider>
-  )
-}
-
-const TriggerElementContext = createContext<ReactElement>(null as any)
-const useTriggerElement = () => useContext(TriggerElementContext)
-
-const Root = menuify(({ children }: MenuRootProps) => {
-  const isNested = useIsNestedMenu()
-
-  if (isNested) {
-    return <MenuProvider>{children}</MenuProvider>
+import { View } from 'react-native'
+
+const createAndroidMenu = (Menu: 'ContextMenu' | 'DropdownMenu') => {
+  const Trigger = menuify(({ children, style }: MenuTriggerProps) => {
```

Comment:
I wonder if we should extract all these placeholder components into a single `native` file to avoid duplicating them for iOS and Android. Maybe for another PR...

Reply:
yeah, thought about it. Another PR should be good!